### PR TITLE
Feature: add https to link filtering

### PR DESCRIPTION
**Pull request by jamesjwarren**

### CODE REVIEW

File: app/js/thisissoon-core/filters/linkDisplay.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ angular.module("thisissoon.core").filter("linkDisplay", [
     "$filter",
     function($filter) {
         return function(input) {
-            return input.replace("http://", "").replace("www.", "");
+            return input.replace("http://", "").replace("https://", "").replace("www.", "");
         };
```
Comment (eopa) [inline app/js/thisissoon-core/filters/linkDisplay.js]: @jamesjwarren would it not be better to use regex for this? something like: `/[a-zA-Z]+:\/\//` would match any protocol and: `/w{3}./`would match `www.`

Comment (jamesjwarren) [inline app/js/thisissoon-core/filters/linkDisplay.js]: good idea

File: app/js/thisissoon-core/filters/linkDisplay.js [hunk 2]
```diff
@@ -9,7 +9,7 @@ angular.module("thisissoon.core").filter("linkDisplay", [
     "$filter",
     function($filter) {
         return function(input) {
-            return input.replace("http://", "").replace("www.", "");
+            return input.replace(/.*?:\/\//g, "").replace(/w{3}./, "").replace(/\/.*/g, "");
         };
```
Comment (eopa) [inline app/js/thisissoon-core/filters/linkDisplay.js]: @jamesjwarren have you tested this? this is a classic case for unit testing

Comment (jamesjwarren) [inline app/js/thisissoon-core/filters/linkDisplay.js]: There are tests for this here: https://github.com/thisissoon/thisissoon-frontend/blob/feature/link-filter/tests/unit/thisissoon-core/filters/linkDisplay.js but I could probably add a few more cases. I'll do that now.

Comment (eopa) [inline app/js/thisissoon-core/filters/linkDisplay.js]: cool, good job :+1:

**Pull request merged**